hamilton/ad_hoc_utils.py

```diff
@@ -60,9 +60,9 @@ def create_temporary_module(*functions: Callable, module_name: str = None) -> Mo
     return module
 
 
-def module_from_source(source: str) -> ModuleType:
+def module_from_source(source: str, module_name: Optional[str] = None) -> ModuleType:
     """Create a temporary module from source code"""
-    module_name = _generate_unique_temp_module_name()
+    module_name = module_name if module_name else _generate_unique_temp_module_name()
```

Comment:
This probably *should* be using a hash of the source. E.G. so we don't repeat it a ton of times. Not sttrictly necessary but feels good.

Reply:
I think it's best to have `module_name` as a required argument. However, didn't want to make that change now to limit the scope of this PR